addons/mail/static/src/core/web/chatter.js

```diff
@@ -269,23 +264,51 @@ export class Chatter extends Component {
     }
 
     /**
+     * Fetch data for the thread according to the request list.
+     * @param {string} threadModel
      * @param {number} threadId
      * @param {['activities'|'followers'|'attachments'|'messages'|'suggestedRecipients']} requestList
      */
     load(
+        threadModel = this.props.threadModel,
         threadId = this.props.threadId,
         requestList = ["followers", "attachments", "messages", "suggestedRecipients"]
     ) {
-        const { threadModel } = this.props;
-        this.state.thread = this.threadService.getThread(threadModel, threadId);
-        this.scrollPosition.model = this.state.thread?.scrollPosition;
+        const thread = this.threadService.getThread(threadModel, threadId);
         if (!threadId) {
             return;
         }
         if (this.props.hasActivities && !requestList.includes("activities")) {
             requestList.push("activities");
         }
-        this.threadService.fetchData(this.state.thread, requestList);
+        this.threadService.fetchData(thread, requestList);
+    }
+
+    /**
+     * @param {string} threadModel
+     * @param {number} threadId
+     * Visually refresh the chatter.
+     */
+    changeThread(threadModel, threadId) {
+        this.state.thread = this.threadService.getThread(threadModel, threadId);
+        this.scrollPosition.model = this.state.thread?.scrollPosition;
+        if (threadId === false) {
+            this.state.composerType = false;
+        }
+        this.attachmentUploader.thread = this.state.thread;
```

Comment:
I'm wondering seeing this line, what happens if there are attachments pending upload during thread change, do they stay correctly linked to the previous thread?

Reply:
The `thread id` should be assigned to the uploader at the beginning of the uploading from the `form data` `    _makeFormData(formData, file, hooker, tmpId, options) {
        formData.append("thread_id", hooker.thread.id);` `            for (const file of files) {
                formData.append("ufile", file);
            }` So I think it should be fine during the pending process.